tests/functional/test_llm_json.py

```diff
@@ -0,0 +1,355 @@
+            "function": {
+                "name": "get_weather",
+                "description": "Get current temperature for a given location.",
+                "parameters": {
+                    "type": "object",
+                    "properties": {
+                        "location": {
+                            "type": "string",
+                            "description": "City and country e.g. Bogotá, Colombia"
+                        }
+                    },
+                    "required": [
+                        "location"
+                    ],
+                    "additionalProperties": False
+                },
+                "strict": True
+            }
+        }]
+
+        messages = [
+            {"role": "user", "content": "What is the weather like in Paris today?"}
+        ]
+        completion = client.chat.completions.create(
+            model=model_name,
+            messages=messages,
+            tools=tools,
+            tool_choice={"type": "function", "function": {"name": "get_weather"}},
+        )
+
+        print("COMPLETION:",completion)
+
+        body = {
+            "model": model_name,
+            "messages": messages,
+            "tools":tools,
+            "tool_choice": {"type": "function", "function": {"name": "get_weather"}}
+        }
+        import requests
+        response = requests.post(
+            url=f"{base_url}/chat/completions",
+            json=body,
+            headers={"Authorization": f"Bearer unused"}
+        )
+
+        if response.status_code == 200:
+            print("API Response:", response.json())
+        else:
+            print(f"Failed to get response: {response.status_code}, {response.text}")
+
+        import json
+        
+        tool_args = completion.choices[0].message.tool_calls[0].function.arguments
+        
+        # Assert that tool_call is a valid JSON and matches the schema
+        import json
+        from jsonschema import validate, ValidationError
+
+        try:
+            tool_call_json = json.loads(tool_args)
+            schema = tools[0]["function"]["parameters"]
+            validate(instance=tool_call_json, schema=schema)
+            assert True, "tool_call is a valid JSON and matches the schema"
+        except json.JSONDecodeError as e:
+            assert False, f"tool_call is not a valid JSON: {e}"
+        except ValidationError as e:
+            assert False, f"tool_call does not match the schema: {e}"
+
+        messages.append({'role': 'assistant', 'reasoning_content': None, 'content': '', 'tool_calls': [{'id': 'chatcmpl-tool-d39b13c90f9b4d48b08c16455553dbec', 'type': 'function', 'function': {'name': 'get_weather', 'arguments': '{"location": "Paris, France"}'}}]})
+        messages.append({"role": "tool", "tool_call_id": "chatcmpl-tool-d39b13c90f9b4d48b08c16455553dbec", "content": "15 degrees Celsius"})
+
+        print("Messages after tool call:", messages)
+
+        completion2 = client.chat.completions.create(
```

Comment:
we could overwrite original completion here I guess... 
completion2 looks a bit awkward